rfcs/chromium-connector.md

```diff
@@ -0,0 +1,213 @@
+# 🚀 Feature request: `connector-chromium` + custom actions
+
+## TL;DR;
+
+Even though we have a `chrome connector` it should probably be renamed to `chromium` as it works with any `chromium` browser that exposes the `chrome DevTools Protocol`. Even `chrome-launcher` supports the environment variable `CHROME_PATH` to indicate the executable of the browser to run.
+
+This alone is not a good reason, so this enoughWhile renaming for the sake of renaming is probably not a good idea, I think we should take this opportunity to modernize the code and create a new connector that will use [`Puppeteer`](https://github.com/GoogleChrome/puppeteer) which was not available when we first started the project.
+
+Using `Puppeteer` will unlock requests like [custom actions](#421) more easily and with ergonomics that developers already know.
+
+## Details
+
+This section contains a list of all the changes and/or new features `connector-chromium`, the CLI, and the Node.js API should implement with some details on how to achieve them.
+
+### User actions
+
+This has been a long standing issue (#421, #1946). The main scenario to unlock is user authentication via the CLI, without forgetting users that need more control of the page before analyzing it. An example of the later will be to click in different elements before triggering the analysis.
+
+The Node.js API should just expose a general way to perform custom actions and the CLI should take advantage of it to allow user authentication.
+
+#### User authentication in the CLI
+
+The authentication steps should be handled by the CLI using the actions ergonomics provided by the Node.js API.
+
+A typical login experience has a user and password `input`s and a submit `button` (everything might or not be inside a `form`). The user fills the inputs, pressed the button, and then the website navigates to another page or updates the URL with a hash. Puppeteer considers a navigation any change in the url, even if it is just a hash so this should cover the most common scenarios.
+
+The information required to do this programatically is:
+
+* **user value** and **`input` selector**
+* **password value** and **`input` selector**
+* **button selector** so it can be `click`ed
+
+In total 5 pieces of information.
+
+The most common ways to provide variables to a CLI are (from lower to higher priority):
+
+1. Environment variables
+1. Configuration file (`.hintrc`)
+1. CLI parameters
+1. A mix of the above: the user provides the selectors via `.hintrc` but uses environment variables for the credentials
+
+With this in mind, the options would be:
+
+| Entry Point | Values  |
+| ----------- | ------- |
+| ENV         | `WEBHINT_USER` <br> `WEBHINT_PASSWORD` <br> `WEBHINT_USER_SELECTOR` <br> `WEBHINT_PASSWORD_SELECTOR` <br> `WEBHINT_SUBMIT` |
```

Comment:
I think all these items should include `AUTH`, `WEBHINT_AUTH_USER` `WEBHINT_AUTH_PASSWORD` and so on.

Reply:
I don't understand your comment 🤔
This is how the table looks like:

![image](https://user-images.githubusercontent.com/606594/55517894-d1cb0500-5626-11e9-985e-488f835edcd7.png)

Reply:
What I mean is that I would include the word `AUTH` for the ENV variables:
so the env variables would be: WEBHINT_AUTH_USER, WEBHINT_AUTH_PASSWORD, WEBHINT_AUTH_USER_SELECTOR, WEBHINT_AUTH_PASSWORD_SELECTOR, WEBHINT_AUTH_SUBMIT

Reply:
OK, only for the `env` variables or do you want me to change the parameters too?

Reply:
Synced with @sarvaje and only `env`

Reply:
Updated envirnment variable names!